tests/simulators/carla/test_carla.py

```diff
@@ -8,6 +9,412 @@
 )
 
 
+def preprocess_old_blueprint_names(original):
+    d = {}
+    for key, value in original.items():
+        for sub_value in value:
+            d[sub_value] = key
+    return d
+
+
+def model_blueprint(simulator, mapPath, town, modelType, modelName, newModelNames=None):
+    code = f"""
+            param map = r'{mapPath}'
+            param carla_map = '{town}'
+            param time_step = 1.0/10
+
+            model scenic.simulators.carla.model
+            ego = new {modelType} with blueprint '{modelName}'
+            terminate after 1 steps
+        """
+    scenario = compileScenic(code, mode2D=True)
+    scene, _ = scenario.generate(maxIterations=10000)
+    simulation = simulator.simulate(scene)
+    obj = simulation.objects[0]
+    if newModelNames:
+        assert obj.blueprint == newModelNames[modelName]
+    else:
+        assert obj.blueprint == modelName
+
+
+from scenic.simulators.carla.blueprints import carModels
+
+
+@pytest.mark.skip(reason="Skipping test due to Carla memory leak issues")
+@pytest.mark.parametrize("modelName", carModels)
+def test_car_blueprints(getCarlaSimulator, launchCarlaServer, modelName):
+    pytest.importorskip("carla")
```

Comment:
Let's not keep repeating this check. We have a whole folder for CARLA tests, so let's use it: you can make `test_blueprints.py`, etc. to organize things better. Then you can use `pytest.skip` to skip entire modules which require CARLA when it's not available. (Alternatively, once the crashing problem is fixed we could make `getCarlaSimulator` a module-scoped fixture.)

Reply:
I was thinking of doing that originally but was unsure if we wanted to restrict all tests into a single file. Will do!